core/dispatcher/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -461,6 +451,8 @@ class Invoker(
         val duration = Duration.fromNanos(java.time.Duration.between(interval._1, interval._2).toNanos)
         val activationResponse = if (duration >= timeout) {
             ActivationResponse.applicationError(s"action exceeded its time limits of ${timeout.toMillis} milliseconds")
+        } else if (log == JsArray("log limit exceeded".toJson)) {
```

Comment:
Yep, both ways are certainly possible. It's a matter of deciding which way we want to go. I agree that the result is independent. One could also think of it as a measure of penalty against a misbehaving action.